src/erlcloud_sns.erl

```diff
@@ -580,12 +580,13 @@ publish(Type, RecipientArn, Message, Subject, Attributes, Config) ->
             Subject -> [{"Subject", Subject}]
         end,
     AttributesParam = message_attributes(Attributes),
-    Doc =
-        sns_xml_request(
+    case sns_xml_request(
             Config, "Publish",
-            RecipientParam ++ MessageParams ++ SubjectParam ++ AttributesParam),
-    erlcloud_xml:get_text(
-        "PublishResult/MessageId", Doc).
+            RecipientParam ++ MessageParams ++ SubjectParam ++ AttributesParam) of
+        {ok, Doc} ->
```

Comment:
This is no longer valid... `sns_xml_request` returns `Doc` or raises an error and every other use in the module appears to treat it this way. I think this `case` change can be undone completely.

Reply:
removed case. Back to original code here